vyper/ast/nodes.py

```diff
@@ -196,13 +251,52 @@ def __init__(self, parent: Optional["VyperNode"] = None, **kwargs: dict):
         if parent is not None:
             parent._children.add(self)
 
+    @classmethod
+    def from_node(cls, node: "VyperNode", **kwargs) -> "VyperNode":
+        """
+        Return a new VyperNode based on an existing node.
+
+        This method creates a new node with the same source offsets as an existing
+        node. The new node can then replace the existing node within the AST.
+        Preserving source offsets ensures accurate error reporting and source
+        map generation from the modified AST.
+
+        Arguments
+        ---------
+        node: VyperNode
+            An existing Vyper node. The generated node will have the same source
+            offsets and ID as this node.
+        **kwargs : Any
+            Fields and values for the new node.
+
+        Returns
+        -------
+        Vyper node instance
+        """
+        ast_struct = {i: getattr(node, i) for i in VyperNode.__slots__ if not i.startswith('_')}
+        ast_struct.update(ast_type=cls.__name__, **kwargs)
+        return cls(**ast_struct)
+
+    @classmethod
+    def get_slots(cls) -> set:
```

Comment:
"slots" seems like a more technical name than we need to expose in this API. Why not `get_fields` instead?

Reply:
:+1: `get_fields` seems much clearer and less scary sounding.  Will change.